scripts/us_eia/opendata/process/coal.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""EIA Coal Dataset specific functions."""
+
+import common
+import logging
+import re
+
+
+def extract_place_statvar(series_id, counters):
+    """Given the series_id, extract the raw place and stat-var ID.
+
+    Args:
+        series_id: EIA series ID
+        counters: map for updating error statistics
+
+    Returns a (place, raw-stat-var, is_us_place) tuple.
+    """
+    # Pattern #1: COAL.{Measure}.{Region}-{Code}.{Period}
+    # Region could include 3-leter codes (e.g. MAT - Middle Atlantic)
+    m = re.match(r"^COAL\.([^._]+_?[^._]+)\.([A-Z]+)-([0-9]+)\.([AQM])$",
+                 series_id)
+    if m:
+        measure = m.group(1)
+        place = m.group(2)
+        code = m.group(3)
+        period = m.group(4)
+        sv_id = f'COAL.{measure}.{code}.{period}'
+        return (place, sv_id, True)
```

Comment:
For context "is_us_place" exists because some datasets have 2 letter country codes which conflict with state code.  In this case, what all can place be?

Reply:
yup, these are all US places. we'll only have intl places for the export/import ones which we'll also need to resolve somehow (since only either dest/src are intl)